### ISSUE TITLE
swiperight , swipleft not working on IE 10 windows 7 Touch Laptop

### ISSUE THREAD

**Issue by vijayphale** [benmajor/jQuery-Touch-Events#68]
Hello Ben,

This library is highly useful. Just used it in one of my page to support for swiperight and swipeleft event and subsequent animation. Its working great on non-touch desktop, laptop, galaxy tab, phone, windows phone, ipad, iphone but failing on Touch laptop IE 10 (windows 7) and IE 11 (windows 8). Nothing happens on swiping on the DOM element. But swipeleft/right event gets fired if i again touch the same element. seems like touchstart and touchend are not getting executed properly.  On the same HTML page, I have implemented Owl Carousel (http://owlgraphic.com/owlcarousel/). That plugin works correctly on touch laptop's IE browser. Could you please let me know how should I make it working on touch laptop

**Comment by benmajor** [comment]
Hi Vijay,

Thanks for your message. Unfortunately, I think that the code to detect the correct event binding is not working in IE on touch-screen enabled devices. I will try to come up with a solution, but the code should still execute using `mousemove` (unless IE is exclusively triggering touch events, which seems strange).

Unfortunately, I do not have a touch-enabled laptop to try this with. I will do some digging and update as and when I can.

Thanks,
Ben.

**Comment by vijayphale** [comment]
Hello Ben,

Thanks for your reply. You are correct, in Touch laptop IE; touch related events are getting executed with type `MSPointerDown` , `MSPointerUp` and 'MSPointerMove'. Hence its not getting recognized by your code.
I have made modifications at few places in your JS file which are as follows:

1) Added new function as : 
```
function getTouches(e) {
        var origEvent = e.originalEvent;
        var touchData = {
            'position': {
                'x': window.navigator.msPointerEnabled ? origEvent.screenX : ((settings.touch_capable) ? origEvent.touches[0].screenX : e.screenX),
                'y': window.navigator.msPointerEnabled ? origEvent.screenY : (settings.touch_capable) ? origEvent.touches[0].screenY : e.screenY
            },
            'offset': {
                'x': window.navigator.msPointerEnabled ? origEvent.offsetX : (settings.touch_capable) ? origEvent.touches[0].pageX - origEvent.touches[0].target.offsetLeft : e.offsetX,
                'y': window.navigator.msPointerEnabled ? origEvent.offsetY : (settings.touch_capable) ? origEvent.touches[0].pageY - origEvent.touches[0].target.offsetTop : e.offsetY
            },
            'time': Date.now(),
            'target': e.target
        };
        return touchData;
    }
```
here I have added the condition for checking `window.navigator.msPointerEnabled ? origEvent.screenX:`

I have called this function from within all the event handlers like `$.event.special.tapstart`, `$.event.special.tapmove` and so on; wherever we need the touch data.

2) modified the event declaration:

```
startevent: window.PointerEvent ? 'pointerdown' : window.navigator.msPointerEnabled ? 'MSPointerDown' : 'touchstart mousedown',
endevent: window.PointerEvent ? 'pointerup' : window.navigator.msPointerEnabled ? 'MSPointerUp' : 'touchend mouseup',
moveevent: window.PointerEvent ? 'pointermove' : window.navigator.msPointerEnabled ? 'MSPointerMove' : 'touchmove mousemove',
```

I did this because on touch laptop for example I can fire `swiperight` event by both ways like by clicking and dragging the mouse OR i can do it via my finger by swiping onto the screen. So basically we need to listen for both `touch` and `mouse` event.

3) Added the `style="-ms-touch-action: none;"` to the element to which i wanted to add the swipe event. Otherwise MS Pointer Event doesn't get fired.

After doing these changes I am now able to interact to DOM element on touch laptops on Internet Explorer 10 and 11

**Comment by benmajor** [comment]
Thanks for looking into this for me, @vijayphale.

I believe that the issue is now fixed by defining the numerous IE specific events inside of the initial declaration. I did not opt to add the additional function call, but instead modified the behaviour of `touch_capable` to return false if it detects `msPointerEnabled` inside of the `navigator` object.

This bug should now be fixed with 1.0.1:

https://github.com/benmajor/jQuery-Touch-Events/tree/master/src/1.0.1
